addon/index.js

```diff
@@ -25,8 +25,19 @@ window.addEventListener('storage', function ({ key, newValue }) {
   localStorageCache.set(key, jsonParseAndFreeze(newValue));
 });
 
-export default function localStorageDecorator(customLocalStorageKey) {
-  return function (target, key, descriptor) {
+export default function localStorageDecoratorFactory(...args) {
+  const isDirectDecoratorInvocation = isElementDescriptor(...args);
+  let customLocalStorageKey = null;
+
+  if (isDirectDecoratorInvocation) {
+    let [target, propertyKey, descriptor] = args;
+    return localStorageDecorator(target, propertyKey, descriptor);
```

Comment:
I think we could just pass the args through, couldn't we?


```suggestion
    return localStorageDecorator(...args);
```

Reply:
Yup! I just thought naming the args would make it more obvious what was happening.